### PR TITLE
ref(*): make log output to stdout by default

### LINKED ISSUE #928: Make log output to stdout default
When running Spin locally in a development loop, I always use `--follow-all` to have my logs written out to the console.

From conversations I've heard others not discovering this option, e.g.:

> Debugging in spin was an initial pain point until finding the log directory. Did not find about follow-all until pretty late

I'd like to propose that writing logs to stdout becomes the default behavior, and that a new option to only output to log files gets introduced, e.g. `--silent` or `--no-stdout`.

### PR DESCRIPTION
**Pull request by michelleN**
+ adds --silent flag to silence logs if needed
+ resolves #928

Signed-off-by: Michelle Dhanani <michelle@fermyon.com>

### CODE REVIEW

File: examples/spin-timer/readme.md [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 To test:
 
-* `cargo build --release`
-* Copy: `cp ./target/release/trigger-timer .`
-* `tar czvf trigger-timer.tar.gz trigger-timer`
-* Update the plugin manifest (`trigger-timer.json`):
-  * Get the SHA: `shasum -a 256 trigger-timer.tar.gz` and copy it into the `sha256` field
-  * Update the URL too, to reflect the directory where the tar file is
-* `spin plugin install --file ./trigger-timer.json --yes`
+- `cargo build --release`
```
Comment (itowlson) [inline examples/spin-timer/readme.md:5]: Is there a reason to change these?  (I don't have an opinion about the change - just surprised to see it in this PR.)

File: crates/trigger/src/cli.rs [hunk 2]
```diff
@@ -64,12 +64,12 @@ where
...
 
-    /// Print all component output to stdout/stderr
+    /// Silence all component output to stdout/stderr
     #[clap(
-        long = "follow-all",
+        long = "silent",
         conflicts_with = FOLLOW_LOG_OPT,
         )]
-    pub follow_all_components: bool,
+    pub silence_component_logs: bool,
```
Comment (itowlson) [inline crates/trigger/src/cli.rs:72]: I wonder if we are at the stage where `--follow-all` is muscle memory enough for people that we should continue to allow the flag as a "no op and deprecation warning."  Not a blocker, just raising it as a thought.

File: crates/trigger/src/cli.rs [hunk 3]
```diff
@@ -64,12 +64,12 @@ where
         )]
     pub follow_components: Vec<String>,
 
-    /// Print all component output to stdout/stderr
+    /// Silence all component output to stdout/stderr
     #[clap(
-        long = "follow-all",
+        long = "silent",
```
Comment (itowlson) [inline crates/trigger/src/cli.rs:69]: I'm a bit unsure about the name here - this promises _silence_, but users will still see Spin output such as "Available routes" links, and Spin logs per the Rust environment variable (errors by default).  I don't have any more constructive suggestions, though, sorry.

**Review by itowlson — APPROVED** [review]
I'd like to bikeshed the "suppress component logs" option name before we merge, but if we don't come up with better ideas then LGTM.  Thank you for cutting through this blasted Gordian knot.

File: examples/spin-timer/readme.md (hunk 1 above)
Comment (michelleN) [reply examples/spin-timer/readme.md]: updated. good catch. think that was some linter changing things for me.

File: crates/trigger/src/cli.rs (hunk 3 above)
Comment (michelleN) [reply crates/trigger/src/cli.rs:69]: `--silence-component-logs` ?

Comment (michelleN) [reply crates/trigger/src/cli.rs:69]: with a short flag `-sh` like 🤫  ??

Comment (itowlson) [reply crates/trigger/src/cli.rs:69]: `-sh` got a giggle out of me!

The long form might be _too_ long, though, and is a bit disconnected from the `follow` terminology used for following single components.  I am wondering about `follow-none` but I can't make myself love it.

Comment (itowlson) [reply crates/trigger/src/cli.rs:69]: (for the avoidance of doubt, "got a giggle out of me" was intended to convey that I really liked it)

Comment (michelleN) [reply crates/trigger/src/cli.rs:69]: How do you feel about `--silence-components`?

Comment (michelleN) [reply crates/trigger/src/cli.rs:69]: `no-app-logs`?
`no-component-logs`?

I'd really love to use `sh`

Comment (itowlson) [reply crates/trigger/src/cli.rs:69]: In isolation I'd probably go for `silence-components`.  But `no-{app|component}-logs` has the advantage that we could rename `follow` to `{app|component}-log` so you could have the related names we had with the `follow` family.

Let's wake @mikkelhegn up and make him have an opinion...

File: crates/trigger/src/cli.rs (hunk 2 above)
Comment (michelleN) [reply crates/trigger/src/cli.rs:72]: Good thought. Wonder if this will clutter peoples logs.

File: crates/trigger/src/cli.rs (hunk 3 above)
Comment (radu-matei) [reply crates/trigger/src/cli.rs:69]: Ok, hear me out: proposal I'm not feeling strongly about, so feel free to ignore.

The default (`spin up` with no flag outputs Spin logs + component logs to stdout / stderr.

`--log-files` / `--log-dir` / `--component-logs-<file-or-dir?>` sends the component logs to files / directory, and the Spin output (available routes, Spin errors) would still go to Spin's stdout / stderr.

Would something like this solve our issue?

File: crates/trigger/src/cli.rs (hunk 2 above)
Comment (michelleN) [reply crates/trigger/src/cli.rs:76]: Added a deprecation notice. Please lmk what you think.

File: crates/trigger/src/cli.rs (hunk 3 above)
Comment (michelleN) [reply crates/trigger/src/cli.rs:73]: I think there is a world where someone may want to see the logs in the output AND save logs to a directory.

Comment (itowlson) [reply crates/trigger/src/cli.rs:73]: If I've understood that right, the way to turn off component log output would be to specify `--log-dir`.  For me that is extremely non-obvious, sorry.  It also leaves us with the "follow only one" problem.

File: crates/trigger/src/cli.rs (hunk 2 above)
Comment (itowlson) [reply crates/trigger/src/cli.rs:76]: Looks good - thanks!

File: crates/trigger/src/cli.rs [hunk 4]
```diff
@@ -179,9 +183,12 @@ where
 
     pub fn follow_components(&self) -> FollowComponents {
         if self.follow_all_components {
-            FollowComponents::All
-        } else if self.follow_components.is_empty() {
+            println!("NOTE: --follow-all has been deprecated. All component ouput is now printed to stdout by default.")
```
[the remark this answers is not in the document]
Comment (michelleN) [reply crates/trigger/src/cli.rs]: updated

File: crates/trigger/src/cli.rs (hunk 2 above)
Comment (michelleN) [reply crates/trigger/src/cli.rs:76]: Should we remove the deprecation notice before cutting 1.0?

Comment (itowlson) [reply crates/trigger/src/cli.rs:76]: Yes, that would be a 4-6 week timeframe, long enough for people who use it routinely enough to have muscle memory to adjust.

File: crates/trigger/src/cli.rs (hunk 3 above)
Comment (lann) [reply crates/trigger/src/cli.rs:73]: `--quiet`/`-q` (no component logs) and `--quieter`/`-qq` (no spin logs either)

**Review by itowlson — APPROVED** [review]

**Pull request merged**